core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/HttpWorkspaceManagerDAO.scala

```diff
@@ -0,0 +1,53 @@
+package org.broadinstitute.dsde.rawls.dataaccess.workspacemanager
+
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import bio.terra.workspace.api.WorkspaceApi
+import bio.terra.workspace.client.ApiClient
+import bio.terra.workspace.model.{CreateDataReferenceRequestBody, CreateWorkspaceRequestBody, CreatedWorkspace, DataReferenceDescription, WorkspaceDescription}
+import org.broadinstitute.dsde.rawls.model.UserInfo
+import spray.json.JsObject
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class HttpWorkspaceManagerDAO(baseWorkspaceManagerUrl: String)(implicit val system: ActorSystem, val materializer: Materializer, val executionContext: ExecutionContext) extends WorkspaceManagerDAO {
+
+  private def getApiClient(accessToken: String): ApiClient = {
+    val client: ApiClient = new ApiClient()
+    client.setBasePath(baseWorkspaceManagerUrl)
+    client.setBearerToken(accessToken)
+
+    client
+  }
+
+  private def getWorkspaceApi(userInfo: UserInfo): WorkspaceApi = {
+    new WorkspaceApi(getApiClient(userInfo.accessToken.token))
+  }
+
+  override def getWorkspace(workspaceId: UUID, userInfo: UserInfo): Future[WorkspaceDescription] = {
+    Future {
+      getWorkspaceApi(userInfo).getWorkspace(workspaceId.toString)
+    }
+  }
+
+  override def createWorkspace(workspaceId: UUID, userInfo: UserInfo): Future[CreatedWorkspace] = {
+    Future {
+      getWorkspaceApi(userInfo).createWorkspace(new CreateWorkspaceRequestBody().id(workspaceId).authToken(userInfo.accessToken.token))
```

Comment:
to replicate what folder manager will eventually do, could this pass a Rawls SA token in the auth header, with the userInfo token in the body?